### PR TITLE
add vault integration

### PR DESCRIPTION
**Pull request by matifali**
- [ ] Test manually
- [ ] Add tests

### CODE REVIEW

File: vault/main.tf [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  validation {
+    condition     = can(regex("^(latest|[0-9]+\\.[0-9]+\\.[0-9]+)$", var.vault_cli_version))
+    error_message = "Vault version must be in the format 0.0.0 or latest"
+  }
+}
+
+resource "coder_script" "vault" {
+  agent_id     = var.agent_id
+  display_name = "vault"
+  icon         = "/icon/vault.svg"
```
Comment (stirby) [inline vault/main.tf:43]: Have you tested this icon? Those outside coder/coder/site/static haven't rendered in my tests.

**Review by stirby — APPROVED** [review]
I added a small note on the icon, plan to test it myself as well. Looks great.

File: vault/main.tf (hunk 1 above)
Comment (matifali) [reply vault/main.tf:43]: İ have a PR to add to there.

Comment (matifali) [reply vault/main.tf:43]: It should render now PR is merged.

Comment (stirby) [reply vault/main.tf:43]: Great. If you want to add tests before merging go ahead, or I can add later.

Comment (matifali) [reply vault/main.tf:43]: Sure. I am going to merge it without tests. İt would be great if you can add tests.

File: vault/run.sh [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+# Verify Vault address and token
+printf "🔎 Verifying Vault address and token ...\n\n"
+vault status
+
+# Store token in .vault-token
+printf "\nStoring token in .vault-token ...\n"
+echo "${VAULT_TOKEN}" > ~/.vault-token
+
+# Add VAULT_ADDR to shell login scripts if not already present e.g. .bashrc, .zshrc
```
Comment (bpmct) [inline vault/run.sh:46]:
```suggestion
# Add VAULT_ADDR to shell login scripts if not already present e.g. .bashrc, .zshrc
# This is a temporary fix and will be replaced with https://github.com/coder/coder/issues/10166
```

File: vault/README.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+### Configure Vault integration and install a specific version of the Vault CLI
+
+```hcl
+module "vault" {
+    source = "https://registry.coder.com/modules/vault"
+    vault_addr = "https://vault.example.com"
+    vault_cli_version = "1.15.0"
+}
+```
```
Comment (bpmct) [inline vault/README.md:47]: Could we add some basic docs on how to fetch a secret with the CLI? I know you had a plan to automatically fetch them all, but I think a very basic example would be helpful, even if it is not formatted right.

**Review by bpmct — APPROVED** [review]

**Comment by matifali** [comment]
As per Vault [docs](https://developer.hashicorp.com/vault/docs/concepts/oidc-provider#public:~:text=An%20access%20token,the%20requesting%20client.) the access_token can only be used to fetch [userinfo](https://developer.hashicorp.com/vault/api-docs/secret/identity/oidc-provider#userinfo-endpoint).

**Pull request closed without merging**